---
version: "3.8"

services:
  nautobot:
    environment:
      - "NAUTOBOT_DB_ENGINE=django.db.backends.mysql"
    env_file:
      - "development.env"
      - "creds.env"
      - "development_mysql.env"
  worker:
    environment:
      - "NAUTOBOT_DB_ENGINE=django.db.backends.mysql"
    env_file:
      - "development.env"
      - "creds.env"
      - "development_mysql.env"
  db:
    image: "mysql:8"
    command:
      - "--default-authentication-plugin=mysql_native_password"
    env_file:
      - "development.env"
      - "creds.env"
      - "development_mysql.env"
    volumes:
      - "mysql_data:/var/lib/mysql"
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: "20s"
      retries: 10
volumes:
  mysql_data: {}
